spec/controllers/stats_spec.rb

```diff
@@ -0,0 +1,57 @@
+describe StatsController do
+  context 'When has valid data' do
+    before do
+      FactoryGirl.reload
+      @program_cc = FactoryGirl.create(
+        :program, name: 'CIÊNCIAS DA COMPUTAÇÃO (BACHARELADO) - CI/João Pessoa')
+      5.times { FactoryGirl.create(:student, :code_sequence, program: @program_cc) }
+      @program_ec = FactoryGirl.create(
+        :program, name: 'ENGENHARIA DE COMPUTAÇÃO (BACHARELADO) - CI/João Pessoa')
+      3.times do
+        FactoryGirl.create(:student, :code_sequence, program: @program_ec, average_grade: 5.5)
+      end
+    end
+
+    it 'is programs count equals to 5 and 3' do
+      get :students
+      expect(assigns(:programs_count)).to include(@program_cc => 5)
+      expect(assigns(:programs_count)).to include(@program_ec => 3)
+    end
+
+    it 'is programs average equals to 5 and 3' do
+      get :students
+      expect(assigns(:programs_average)).to include(@program_cc => 7.34)
```

Comment:
It comes from the default Student on Student Factory. Maybe I should explicit the Student average_grade value in this case...